src/Negotiation/LanguageNegotiator.php

```diff
@@ -2,78 +2,36 @@
 
 namespace Negotiation;
 
-/**
- * @author William Durand <william.durand1@gmail.com>
- */
-class LanguageNegotiator extends Negotiator
+class LanguageNegotiator extends AbstractNegotiator
 {
     /**
-     * {@inheritDoc}
+     * {@inheritdoc}
      */
-    protected function parseHeader($header)
+    protected function acceptFactory($accept)
     {
-        $acceptHeaders = array();
-
-        $header      = preg_replace('/\s+/', '', $header);
-        $acceptParts = array();
-
-        preg_match_all(
-            '/(?<=[, ]|^)([a-zA-Z-]+|\*)(?:;q=([0-9.]+))?(?:$|\s*,\s*)/i',
-            $header,
-            $acceptParts,
-            PREG_SET_ORDER
-        );
-
-        $index    = 0;
-        $catchAll = null;
-        foreach ($acceptParts as $acceptPart) {
-            $value   = $acceptPart[1];
-            $quality = isset($acceptPart[2]) ? (float) $acceptPart[2] : 1.0;
-
-            if ('*' === $value) {
-                $catchAll = new AcceptHeader($value, $quality);
-            } else {
-                $acceptHeaders[] = array(
-                    'item'  => new AcceptHeader($value, $quality),
-                    'index' => $index
-                );
-            }
-
-            $index++;
-        }
-
-        return $this->sortAcceptHeaders($acceptHeaders, $catchAll);
+        return new AcceptLanguage($accept);
     }
 
     /**
-     * {@inheritDoc}
+     * {@inheritdoc}
      */
-    protected function match(array $acceptHeaders, array $priorities = array())
-    {
-        $wildcardAccept  = null;
+    protected static function match(BaseAccept $acceptLanguage, BaseAccept $priority, $index) {
```

Comment:
I just thought that someone extending this class might want to override it. A new Accept-? header, maybe, or an alternative implementation.